core/src/main/java/com/novoda/noplayer/drm/DrmSessionCreatorFactory.java

```diff
@@ -18,7 +18,7 @@ public DrmSessionCreator createFor(DrmType drmType, DrmHandler drmHandler) {
                 );
                 return StreamingDrmSessionCreator.newInstance(mediaDrmCallback, new FrameworkMediaDrmCreator());
             case WIDEVINE_MODULAR_DOWNLOAD:
-                return new NoDrmSessionCreator();
+                return new DownloadDrmSessionCreator((DownloadedModularDrm) drmHandler, new FrameworkMediaDrmCreator());
```

Comment:
it's because the lifecycle of the player is 

```
create a player -> for content and drm type and then attach listeners
load a video -> create and start drm session passing in previously set up listeners
```